blockchain/chain/src/store/chain_store.rs

```diff
@@ -539,26 +543,33 @@ where
 
     /// Subscribes to head changes. This function will send the current tipset through a channel,
     /// start a task that listens to each head change event and forwards into the channel,
-    /// then returns the receiver of this channel from the function.
+    /// then returns an id corresponding to the receiver of this channel from the function.
     /// This function is not blocking on events, and does not stall publishing events as it will
     /// skip over lagged events.
-    pub async fn sub_head_changes(&self) -> Receiver<HeadChange> {
-        let (tx, rx) = bounded(16);
+    pub async fn sub_head_changes(&self) -> i64 {
+        let (tx, rx) = channel::unbounded();
```

Comment:
Should probably not be unbounded...

Reply:
What's wrong with unbounded?

Reply:
If the websocket connection is slow, then we might end up just filling up the unbounded buffer. Is there a reason not to bound this by something reasonable like 16 (what we had before)? Seems like limiting will only benefit us if we ever have to chase down memory issues

Reply:
Alright, I've put it back to a bounded channel.